### PR TITLE
chore: update http-client

### PR DESCRIPTION
**Pull request by cristianvasquez**
Hi @bergos, 

In this branch, I'm trying to update sparql-http-client to version "^2.4.1", but I'm having trouble getting the status code from the endpoint to forward it. 

What would be a good way to do so?

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -78,13 +78,13 @@ export class SparqlHandler {
...
     try {
-      const stream = await this.streamClient.query.construct(query, {headers:queryOptions})
+      const stream = await this.streamClient.query.construct(query, { headers: queryOptions })
       return {
-        status: 200, stream,
+        status: 200, stream
       }
     } catch (error) {
       return {
-        status: error.status,
```
Comment (cristianvasquez) [inline index.js]: This is the part I'm unsure about. I inspected the code of http-server-client, and it throws an error object with {status},  would it be more cases?

**Comment by bergos** [comment]
You should use the `SimpleClient`. It will return the raw `fetch` `Response` object. Then you have access to all required fields. All other implementations would parse the response, which we don't even want in this package.

File: index.js [hunk 2]
```diff
@@ -61,50 +62,58 @@ export class SparqlHandler {
...
+        return this.get(req, res, next, req.iri)
     }
+
+    return next()
+  }
+
+  async head (_req, res, next, iri) {
+    iri = encodeURI(iri)
+
+    debug('handle OPTIONS request for IRI <' + iri + '>')
```
Comment (ludovicm67) [inline index.js]: I noticed that I forgot to replace this `OPTIONS` by `HEAD` in this log line.
Can you take care of it? :)

```suggestion
    debug('handle HEAD request for IRI <' + iri + '>')
```

Comment (cristianvasquez) [reply index.js]: Sure!, I've updated it now.

**Comment by cristianvasquez** [comment]
@bergos I've updated the client to SimpleClient as requested. However, this change breaks https://github.com/zazuko/trifid-renderer-entity/blob/main , that now returns TTL as default instead of HTML . 

I'm uber-confused about what needs to be set and what header. I need a sequence diagram on how trifid calls all the stack.  (or another person that sets the headers)

File: index.js (hunk 2 above)
Comment (cristianvasquez) [reply index.js]: Updated!

File: index.js [hunk 3]
```diff
@@ -123,17 +134,17 @@ export class SparqlHandler {
...
+      const { status, stream } = await this.graphStream(iri, query,
+        req.headers.accept)
 
       if (!stream) {
         return next()
       }
       res.status(status)
-      Object.keys(headers).forEach(name => {
-        res.setHeader(name, headers[name])
-      })
```
Comment (bergos) [inline index.js]: I don't think you can remove this. The SPARQL endpoint sends the `content-type`, and maybe some other headers that must be forwarded.

File: package.json [hunk 4]
```diff
@@ -2,7 +2,7 @@
   "name": "trifid-handler-sparql",
   "description": "SPARQL handler for Trifid",
   "type": "module",
-  "version": "2.0.3",
```
Comment (bergos) [inline package.json:5]: The version should only be changed with `npm version`.

**Review by bergos — CHANGES_REQUESTED** [review]
The missing headers could have caused the described problems.

**Comment by bergos** [comment]
> @bergos I've updated the client to SimpleClient as requested. However, this change breaks https://github.com/zazuko/trifid-renderer-entity/blob/main , that now returns TTL as default instead of HTML .

The link doesn't work for me.

File: package.json (hunk 4 above)
Comment (ludovicm67) [reply package.json:5]: +1

And this should never be changed in a branch other than the `main` one.

**Comment by cristianvasquez** [comment]
> > @bergos I've updated the client to SimpleClient as requested. However, this change breaks https://github.com/zazuko/trifid-renderer-entity/blob/main , that now returns TTL as default instead of HTML .
> 
> The link doesn't work for me.

Sorry I wrote pasted the wrong link,  https://github.com/zazuko/trifid-renderer-entity/blob/main/index.js

**Comment by tpluscode** [comment]
Please recreate this PR is needed. This one appears to introduce a lot of unnecessary formatting changes and somehow GitHub wants to merge commits which are already present on the master branch

**Pull request closed without merging**